ROX-22480: Changelog entry for future hardening (#9903)
stackrox/stackrox: CHANGELOG.md

    - Secret, Secret Path
- The following search terms will be disabled in the next release and removed from the secret context in 2 releases. They can be removed in the current release by setting ROX_SECRET_FILE_SEARCH=false:
  - Secret Type, Cert Expiration,Image Pull Secret Registry

> - The `/v1/availableAuthProviders` endpoint will soon require authentication and at least READ permission on the `Access` resource.

Reviewer: ```suggestion
- The `/v1/availableAuthProviders` endpoint will in a future release require authentication and at least READ permission on the `Access` resource.
```

>   Ensure that any flow interacting with it is authenticated and has the proper permissions going forward.

Reviewer 2: I think we prefer `future` over `soon` at least in this file.

> - The `/v1/tls-challenge` will soon require authentication, ensure that any flow interacting with these endpoints is authenticated going forward.

Reviewer 2: I'm not an expert but I think flow is not the word we are looking for
```suggestion
- The `/v1/tls-challenge` will soon require authentication. Ensure that all interactions with these endpoints include proper authentication
```
